### ISSUE TITLE
Error building Mist >> Thread 1: signal SIGABRT

### ISSUE THREAD

**Issue by perez987** [ninxsoft/Mist#39]
Good afternoon. I really like Mist and use the release version to my satisfaction.
But when I try to compile it from source, I get the error `Thread 1: signal SIGABRT`. Mist icon keeps bouncing in the Dock but the app never quite starts up.
I'm running Xcode 14.1, swift 5.7 and macOS Ventura.

<details>

<summary>Mist error</summary><br>

<img width="1073" alt="Mist-error" src="https://user-images.githubusercontent.com/50429399/225415568-bba43406-65b0-4cc6-8c42-e888bb1b97f0.png">

</details>

**Comment by ninxsoft** [comment]
@perez987 Requesting some additional information to help isolate the cause of this issue:

- In the dubug tab of the navigator (currently selected in your screenshot), are you able to provide more details on the `Thread 1` that's crashing? A larger screenshot or a way to export the stacktrace perhaps.
- Does this also occur on **Xcode 14.2** or **14.3**, **Swift 5.8** and **macOS Ventura 13.3.1**?
- Does this crash only occur on your Mac? Do you have a second Mac / VM you can test this on?
- How about [mist-cli](https://github.com/ninxsoft/mist-cli) ? Perhaps there's some common or differing configuration between the two projects that I might be able to isolate.

**Comment by perez987** [comment]
@ninxsoft thanks, I’m not at home, I’ll check it and comment.

**Comment by perez987** [comment]
@ninxsoft Good night.

I've updated Xcode to 14.3 (swift is 5.8 now). Getting mist project from scratch. Same errors. It seems to be building the app but the main windows never is showed. Instead, I get errors in:

- output console: related to Sparkle although this package seems to be well downloaded and integrated into the project
- Thread 1 >> 5 start: `->  0x7ff8084a93bd <+1805>: movq   %rax, %rbx Thread 1: signal SIGABRT`
- Thread 1 >> 0__abort-with_payload: `->  0x7ff80850ec42 <+10>: jae    0x7ff80850ec4c  ; <+20> Thread 1: signal SIGABRT`. 

The 3 texts zipped and attached. It's only for curiosity since Mist app downloaded from releases page works very well.

[errors.zip](https://github.com/ninxsoft/Mist/files/11244737/errors.zip)

**Comment by ninxsoft** [comment]
@perez987 Thank you for the error logs, I believe I have been able to replicate the issue!

You were on the right track, Sparkle is the culprit. More specifically, Sparkle requires signing certificates in order to work correctly.

If you have access to an **Apple Developer ID** and a **Signing Certificate**, you can change the target settings to specify a team and signing certificate:

![Signing](https://user-images.githubusercontent.com/4812595/233769727-7fa1aa06-9ff1-4cd4-b82b-a775bd0f4527.png)

My understanding is that this is a hard requirement for Sparkle to work correctly. The alternative would be to remove Sparkle locally as a dependency and remove all references to the Sparkle framework in code 👍

**Comment by perez987** [comment]
@ninxsoft 
Thank you! I’ll try and comment.

**Comment by perez987** [comment]
@ninxsoft 
Good night. The issue is solved. Indeed, it has to do with signing.

Although the error is mainly from Sparkle, there is no need to do anything in this package. What needs to be done is, as you say, sign the app with an Apple Developer account in the Signing & Capabilities tab:

- Automatically Manage Signing: Yes
- Team: my Developer ID account
- Provisioning Profile: None
- Signing Certificate: Development (Sign to Run Locally gives the error).

<img width="538" alt="mist" src="https://user-images.githubusercontent.com/50429399/233863676-962cb96b-ab4a-4bc1-ab1d-59bbba4a7ddf.png">

In short, without an Apple Developer account  you cannot build Mist.

Thanks for your interest.
